cmd/elastic_ip_show.go

```diff
@@ -106,58 +104,66 @@ func (c *elasticIPShowCmd) cmdPreRun(cmd *cobra.Command, args []string) error {
 }
 
 func (c *elasticIPShowCmd) cmdRun(_ *cobra.Command, _ []string) error {
-	ctx := exoapi.WithEndpoint(gContext, exoapi.NewReqEndpoint(account.CurrentAccount.Environment, c.Zone))
+	ctx := gContext
+	client, err := switchClientZoneV3(ctx, globalstate.EgoscaleV3Client, c.Zone)
+	if err != nil {
+		return err
+	}
 
-	elasticIP, err := globalstate.EgoscaleClient.FindElasticIP(ctx, c.Zone, c.ElasticIP)
+	elasticIPs, err := client.ListElasticIPS(ctx)
 	if err != nil {
-		if errors.Is(err, exoapi.ErrNotFound) {
-			return fmt.Errorf("resource not found in zone %q", c.Zone)
-		}
+		return err
+	}
 
+	elasticIP, err := elasticIPs.FindElasticIP(c.ElasticIP)
+	if err != nil {
 		return err
 	}
 
 	out := elasticIPShowOutput{
-		ID:            *elasticIP.ID,
-		IPAddress:     elasticIP.IPAddress.String(),
-		AddressFamily: utils.DefaultString(elasticIP.AddressFamily, ""),
-		CIDR:          utils.DefaultString(elasticIP.CIDR, ""),
-		Description:   utils.DefaultString(elasticIP.Description, ""),
+		ID:            elasticIP.ID,
+		IPAddress:     elasticIP.IP,
+		AddressFamily: elasticIP.Addressfamily,
+		CIDR:          elasticIP.Cidr,
+		Description:   elasticIP.Description,
 		Zone:          c.Zone,
 		Type:          "manual",
 	}
 
-	rdns, err := globalstate.EgoscaleClient.GetElasticIPReverseDNS(ctx, c.Zone, *elasticIP.ID)
+	rdns, err := client.GetReverseDNSElasticIP(ctx, elasticIP.ID)
 	if err != nil {
-		if errors.Is(err, exoapi.ErrNotFound) {
+		if errors.Is(err, v3.ErrNotFound) {
 			out.ReverseDNS = ""
 		} else {
 			return err
 		}
+	} else {
+		out.ReverseDNS = rdns.DomainName
 	}
 
-	out.ReverseDNS = rdns
-
-	attachedInstances, err := utils.GetInstancesAttachedToEIP(ctx, globalstate.EgoscaleClient, elasticIP.IPAddress.String(), c.Zone)
+	attachedInstances, err := utils.GetInstancesAttachedToEIP(ctx, client, elasticIP.IP)
 	if err != nil {
 		return err
 	}
 
-	for _, instance := range attachedInstances {
-		out.Instances = append(out.Instances, *instance.Name)
+	for _, instance := range attachedInstances.Instances {
+		out.Instances = append(out.Instances, instance.Name)
 	}
 
 	if elasticIP.Healthcheck != nil {
 		out.Type = "managed"
-		out.HealthcheckMode = elasticIP.Healthcheck.Mode
-		out.HealthcheckPort = elasticIP.Healthcheck.Port
-		out.HealthcheckURI = elasticIP.Healthcheck.URI
-		out.HealthcheckInterval = elasticIP.Healthcheck.Interval
-		out.HealthcheckTimeout = elasticIP.Healthcheck.Timeout
-		out.HealthcheckStrikesOK = elasticIP.Healthcheck.StrikesOK
-		out.HealthcheckStrikesFail = elasticIP.Healthcheck.StrikesFail
-		out.HealthcheckTLSSNI = elasticIP.Healthcheck.TLSSNI
-		out.HealthcheckTLSSkipVerify = elasticIP.Healthcheck.TLSSkipVerify
+		// Message for reviewer:
+		// I don't really like to use this function for only one this parameter, if you have a better solution, please feel free to suggest it
+		// TODO: remove comment before merging
```

Comment:
This part is not gorgeous I agree, it's hard to define what's an empty value for certain type if we are not relying on pointers, example for int or bool, probably for sting types, you can remove pointer and rely only omitempty tag fpr empty string.
Probably need to be verified for other types like int, if in this case we can consider 0 as an empty value...etc